code/game/objects/items/robot/robot_items.dm

```diff
@@ -326,6 +340,167 @@
 		cooldown = world.time + 600
 		log_game("[user.ckey]([user]) used an emagged Cyborg Harm Alarm in ([user.x],[user.y],[user.z])")
 
+/obj/item/borg/lollipop
+	name = "Lollipop Fabricator"
+	desc = "Reward good humans with this. Toggle in-module to switch between dispensing and high velocity ejection modes."
+	icon_state = "lollipop"
+	var/candy = 30
+	var/candymax = 30
+	var/charge_delay = 10
+	var/charging = 0
+	var/mode = 1
+	var/firedelay = 0
+	var/hitspeed = 2
+	var/hitdamage = 0
+	var/emaggedhitdamage = 3
+
+/obj/item/borg/lollipop/equipped()
+	check_amount()
+
+/obj/item/borg/lollipop/dropped()
+	check_amount()
+
+/obj/item/borg/lollipop/proc/check_amount()	//Doesn't even use processing ticks.
+	if(charging)
+		return
+	if(candy < candymax)
+		addtimer(src, "charge_lollipops", charge_delay, TIMER_NORMAL)
+		charging = 1
+
+/obj/item/borg/lollipop/proc/charge_lollipops()
+	candy++
+	charging = 0
+	check_amount()
+
+/obj/item/borg/lollipop/proc/dispense(atom/A, mob/user)
+	if(candy <= 0)
+		user << "<span class='warning'>No lollipops left in storage!</span>"
+		return FALSE
+	var/turf/T = null
+	if(isturf(A))

```

Comment:
replace this if check chains with

     var/turf/T =  get_turf(A)
     if(!T || !istype(T))
        return FALSE
     new /obj/item/weapon/reagent_containers/food/snacks/lollipop(T)
    
It should work exactly the same and be much cleaner